### DISCUSSION TITLE
Brainstorming UI elements

### DISCUSSION THREAD

**Discussion by DennaGherlyn** [Hypfer/Valetudo#829] [Archive]
There has been a lot of discussion about a new UI on telegram. This is a collection of what was said so far and for further additions. 

I added links to the relevant part of the discussion in telegram. Note that these links won't work if your not a member of that group yet.

State of the new UI on writing (by @Jomik [Repo](https://github.com/Jomik/Valetudo-UI)):
![photo_2021-04-12_12-44-56](https://user-images.githubusercontent.com/13550824/114386485-90647d00-9b91-11eb-9427-a336d450d7ef.jpg)

- Discussion about the controls [*](https://t.me/c/1357472831/40443)
  - Basic controls should be easily reachable [*](https://t.me/c/1357472831/40450)
    - Controls with a mini map and a separate map view for advanced map functionality like segmentation? [*](https://t.me/c/1357472831/40450)
    - [valetudo mockup by Depau](https://github.com/Hypfer/Valetudo/files/6296105/valetudo.mockup-1.pdf)[*](https://t.me/c/1357472831/40484)
      - "Edit" being the current zones page, "that section could also contain go-to presets / configured zones / segments / whatever makes sense" [*](https://t.me/c/1357472831/40509)
    - Change of the start cleaning symbol based on cleaning selection. Depau: "Maybe once you add segments it may become ▶️📍 to indicate "start segment cleaning"" [*](https://t.me/c/1357472831/40460) 
  - Where to position the stats? [*](https://t.me/c/1357472831/40461)
    - On top of the drawer?
  - Intensity:
    - Instead of sliders for the intensity a dropdown, because the backend has no understanding of more/less. For example, roborocks "off" speed comes after "max". [*](https://t.me/c/1357472831/40456) 
    - Speed dial? [*](https://t.me/c/1357472831/40457)
  - Single page for controls & map [*](https://t.me/c/1357472831/40501)
  - Only settings and advanced stuff (like remote control) on additional pages [*](https://t.me/c/1357472831/40504)
  - Separators between different types of presets/segments [*](https://t.me/c/1357472831/40512)
  - Put labels above sliders and such for mobile friendliness [*](https://t.me/c/1357472831/40516)
  - Elements being accordion style? [*](https://t.me/c/1357472831/40525)
- Battery charge on the top bar (icon & percentage) [*](https://t.me/c/1357472831/40544)
![battery](https://user-images.githubusercontent.com/13550824/114384825-8477bb80-9b8f-11eb-92bd-1a39c10b7c9e.jpg)
- Finding a fitting mop icon [*](https://t.me/c/1357472831/40532)

**Comment by DennaGherlyn** [comment]
**Update 2021-04-13** 
 - Which controls to put on the main page? [*](https://t.me/c/1357472831/40904) [*](https://t.me/c/1357472831/40951)
    - ✅ basic control
    - ☑️ carpet
    - ☑️ restrictions
    - ☑️ consumables
    - ✅ fan speed
    - ☑️ locate (not really necessary)
    - ☑️ manual control
    - ☑️ map reset
    - ☑️ all map segmentation
    - ☑️ persistent map
    - ☑️ volume/test volume
    - ☑️ voice pack
    - ✅ water grade
    - ✅ (in the map, maybe zone presets in panel) zone cleaning
    - ✅ (in the map, maybe gotopresets in panel) goto location
    - ☑️ sensor calibration
    - ☑️ wifi
    - status:
      - ✅ attachments
      - ✅ battery
      - ☑️ consumables
      - ✅ fanspeed/water grade
      - ✅ (in the map, or maybe not, don't care) latest cleanup
      - ✅ status
- Two drawers? Menu and controls? [*](https://t.me/c/1357472831/40969)
- Using horizontal space on Desktop and vertical space on Mobile [*](https://t.me/c/1357472831/41007)
- Positioning of menus:
  - Menu at the top? Like Google Play Store [*](https://t.me/c/1357472831/41083)
    ![gps](https://user-images.githubusercontent.com/13550824/114501540-79be3480-9c2a-11eb-99fd-ecfe4e8bf974.jpg)
    ![gps2](https://user-images.githubusercontent.com/13550824/114501549-7cb92500-9c2a-11eb-86e2-f161296abbdd.jpg)
  - Swipe from the bottom up? [*](https://t.me/c/1357472831/41099)
    ![swipe_up](https://user-images.githubusercontent.com/13550824/114501528-7460ea00-9c2a-11eb-9c4f-735b8ee0ee3e.jpg)
    - Some phone use swipe up for navigation [*](https://t.me/c/1357472831/41106)
      - Not completely from the bottom [*](https://t.me/c/1357472831/41109)

**Comment by Jomik** [comment]
Here is some images of the current state of the UI I am working on.

<details>
<summary>
Show images
</summary>

![image](https://user-images.githubusercontent.com/699655/118408019-e4c4b600-b683-11eb-9b91-31f5b560da9b.png)

![image](https://user-images.githubusercontent.com/699655/118408037-f0b07800-b683-11eb-95fc-c670ab204860.png)

![image](https://user-images.githubusercontent.com/699655/118408046-f60dc280-b683-11eb-9055-e7bdd21e89ad.png)

![image](https://user-images.githubusercontent.com/699655/118408056-01f98480-b684-11eb-8023-82b0b110fab6.png)

![image](https://user-images.githubusercontent.com/699655/118408073-13db2780-b684-11eb-986e-27753a713acc.png)

![image](https://user-images.githubusercontent.com/699655/118408082-2190ad00-b684-11eb-99c3-15af0a8abc6d.png)

![image](https://user-images.githubusercontent.com/699655/118408105-3a995e00-b684-11eb-832c-49515ff6d154.png)

![image](https://user-images.githubusercontent.com/699655/118408093-2e150580-b684-11eb-9713-e265b0c147b4.png)

</details>
